methods/articles.coffee

```diff
@@ -4,20 +4,17 @@ Meteor.methods
   addEventSource: (source) -> #eventId, url, publishDate, publishDateTZ
     user = Meteor.user()
     if user and Roles.userIsInRole(user._id, ['admin'])
-      if source.url
-        insertArticle =
-          url: source.url
-          title: source.title
-          userEventId: source.userEventId
-        existingArticle = Articles.find(
-          userEventId: source.userEventId
-          url: source.url
-          deleted:
-            $in: [null, false]
-        ).fetch()
-        unless existingArticle.length is 0
-          throw new Meteor.Error(501, 'This article has already been added')
-        else
+      # Check if Article is in collection
+      source = Articles.findOne(url: $regex: new RegExp("#{sourceId}$"))
```

Comment:
This could match the wrong article if one has an id that is a substring of another. For example, `10` and `110`.